pkg/virt-controller/watch/migration.go

```diff
@@ -448,12 +454,19 @@ func (c *MigrationController) updateStatus(migration *virtv1.VirtualMachineInsta
 		log.Log.Object(migration).Error("vmi's migration state was taken over by another migration object")
 	} else if vmi.Status.MigrationState != nil &&
 		vmi.Status.MigrationState.MigrationUID == migration.UID &&
-		vmi.Status.MigrationState.Failed {
+		(vmi.Status.MigrationState.Failed || vmi.Status.MigrationState.AbortRequested) {
```

Comment:
I don't think this is correct. Consider a case where migration is running.

Reply:
This isn't meant to treat the case of a running migration, only a migration that did not start yet (either target pod is not created or handoff did not occur yet).

The case of a running migration that's being cancelled already works before this PR.

Reply:
Can you elaborate on how does this not get to affect running migrations?

Reply:
I though the [condition here](https://github.com/kubevirt/kubevirt/pull/7599/files#diff-1baf192bb9c4560d7d40f008b7ba5c680ebce669b8f1b7639445abd3e7269688R1069) prevents it:
```go
migration.DeletionTimestamp != nil && !migration.IsFinal()
```

But the previous condition was:
```go
case virtv1.MigrationRunning:
		// abort the migration if the migration is being deleted.
		if migration.DeletionTimestamp != nil && vmi.Status.MigrationState != nil {
```

So you're right. I need to add `vmi.Status.MigrationState != nil` to the current condition. This would mean if the migration is running we would abort only if handoff did not occur yet, as in the original condition.

Thanks for catching that!

Reply:
Right, it does affect the already running migration and will prematurely finalize the migration. That's why we should rely on the canceled migrations map and set the `vmi.Status.MigrationState.Failed` for migrations that didn't reach the handoff and actively prevent the handoff.

Reply:
Let's remove the `|| vmi.Status.MigrationState.AbortRequested` from here

Reply:
Yes, @xpivarc is right.
Thanks a lot for the important observation.

I've removed the `|| vmi.Status.MigrationState.AbortRequested` part and made tiny adjustments

Reply:
@xpivarc @vladikr 
Please take another look on https://github.com/kubevirt/kubevirt/pull/7599/commits/aac582044a1808244d5887c1036b9587081914e3.

Now there is no race. I've edited `handleSignalMigrationAbort()` to receive another `toMarkAsFailed` parameter. The (only) caller passes `c.isMigrationAborted(migration.UID)` argument.

In other words, now if we signal abortion because `c.isMigrationAborted(migration.UID) == true` it means that no handoff occurred yet, therefore it's safe to fail the migration. Else, we only want to set `MigrationState.AbortRequested` to true and let virt-handler take care of things. Only after it will fail the migration - the migration will be marked as failed and therefore be finalized.